test/metabase/api/card_test.clj

```diff
@@ -2814,14 +2815,27 @@
                  :values_source_config {:values ["BBQ" "Bakery" "Bar"]}}]
                (:parameters card)))))))
 
-(defn- upload-example-csv! [collection-id]
-  (let [file (upload-test/csv-file-with
-              ["id, name"
-               "1, Luke Skywalker"
-               "2, Darth Vader"]
-              "example_csv_file")]
-    (mt/with-current-user (mt/user->id :rasta)
-      (api.card/upload-csv! collection-id "example_csv_file.csv" file))))
+(defn upload-example-csv!
+  "Upload a small CSV file to the given collection ID"
+  ([collection-id]
+   (upload-example-csv! collection-id true))
+  ([collection-id grant-permission?]
+   (let [file        (upload-test/csv-file-with
+                      ["id, name"
+                       "1, Luke Skywalker"
+                       "2, Darth Vader"]
+                      "example_csv_file")
+         group-id    (u/the-id (perms-group/all-users))
+         grant-perms (or (and grant-permission?
+                              (u/ignore-exceptions
+                                (classloader/require 'metabase-enterprise.advanced-permissions.models.permissions)
+                                (resolve 'metabase-enterprise.advanced-permissions.models.permissions/update-db-details-permissions!)))
+                         (fn [_ _ _]))]
+     (grant-perms group-id (mt/id) :yes)
+     (u/prog1
+       (mt/with-current-user (mt/user->id :rasta)
+         (api.card/upload-csv! collection-id "example_csv_file.csv" file))
+       (grant-perms group-id (mt/id) :no)))))
```

Comment:
All of this brouhaha is because:

1. This is a useful function for normal tests, and we don't want every unit test to have to get permissions right for Rasta. But it seemed rude to grant permission when it's normally not given, so I take it back afterwards.
2. It's also used for the permission test in `enterprise/backend/test/metabase_enterprise/advanced_permissions/common_test.clj` where permission is specifically *not* granted, so I had to add the boolean flag.